icu4c/source/i18n/number_longnames.cpp

```diff
@@ -431,13 +431,39 @@ void getMeasureData(const Locale &locale,
     subKey.append(unit.getType(), status);
     subKey.append("/", status);
 
+    // Check if unitSubType is an alias or not.
+    CharString unitSubTypeCharString;
+    unitSubTypeCharString.append(unit.getSubtype(), status);
+
+    LocalUResourceBundlePointer aliasBundle(ures_open(U_ICUDATA_ALIAS, "metadata", &status));
+
+    UErrorCode aliasStatus = status;
+    StackUResourceBundle aliasFillIn;
+    CharString aliasKey;
+    aliasKey.append("alias/unit/", aliasStatus);
+    aliasKey.append(unitSubTypeCharString, aliasStatus);
+    aliasKey.append("/replacement", aliasStatus);
+    ures_getByKeyWithFallback(aliasBundle.getAlias(), aliasKey.data(), aliasFillIn.getAlias(),
+                              &aliasStatus);
+    // This means the subType is an alias. Then, replace unitSubType with the replacement.
+    if (!U_FAILURE(aliasStatus)) {
+        int32_t len;
+        UBool forceCopy = true;
+        unitSubTypeCharString.clear();
+        ures_getUTF8String(aliasFillIn.getAlias(), unitSubTypeCharString.data(), &len, forceCopy,
+                           &aliasStatus);
+    }
+
+    const char *unitSubType = unitSubTypeCharString.data();
+
```

Comment:
@pedberg-icu , I think there is a better way in C++ to get the data for aliases that what I am doing here. I am not familiar with getting the data in C++. I want to ask you if you think there is a better way.

Reply:
> I will think about that a bit. In meetings now for 3 hours. Adding @richgillam as reviewer, he may have good ideas about this

Looking at this again, I do not know of a better way to get the alias data, I would go ahead with this approach.

Reply:
OKay, I will fix the clang asan and then send it back to you

Reply:
thanks :)